### PR TITLE
Magnetometer calibration

### PR DESCRIPTION
**Pull request by jerelbn**
added a magnetometer calibration currently compatible with the mag_mods branch of ROSflight

### CODE REVIEW

File: src/mavrosflight/sensors/mag.cpp [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+             h, b, f,
+             g, f, c;
+
+        Eigen::MatrixXd ub(3, 1);
+        ub << 2 * p,
+              2 * q,
+              2 * r;
+        double k = d;
+
+        // eq. 21 of Renaudin (should be negative according to eq. 16)
```
Comment (superjax) [inline src/mavrosflight/sensors/mag.cpp:151]: Dude, This is awesome!  Thanks for putting in references to papers.

Any chance you could put the full reference in somewhere?

**Review by superjax — APPROVED** [review]
Thanks Jerel!  This is really nice code.

I need to test this on my own naze, but I can't think of anything else besides maybe putting the full reference in somewhere.

I think I also want to put your visualizer in here somewhere, with a flag that chooses whether it is published or not.  I think that would be pretty cool.

**Comment by jerelbn** [comment]
Thanks. I didn't even realize I forgot to also copy the paper name from my nonROS test code... I can put the visualizer somewhere, too.

**Comment by jerelbn** [comment]
Oh, I actually put the paper name in the header file. I will add a full citation above each function in the header and source.

**Pull request merged**